### ISSUE TITLE
Significant(?) difference between SkyCoord.apply_space_motion and other astrometry libraries/tools

### ISSUE THREAD

**Issue by tboch** [astropy/astropy#11009]
### Description

I tested `SkyCoord.apply_space_motion` for a sample of 100,000 Gaia DR2 stars (with significant proper motions and parallaxes) and compared the results with 3 other tools/services: CDS internal _astrocoo_ library (Java), Topcat  _epochProp( tYr, astrom6 )_ function and Gaia ESA ADQL user-defined function _EPOCH_PROP_ (cf. https://gea.esac.esa.int/archive-help/adql/epochprop/index.html )

### Expected behavior

I would expect astropy to give results very close to those obtained with the 3 other methods.
The distance is directly proportional to the total proper motion value, as shown by this plot:

![image](https://user-images.githubusercontent.com/5598168/98578404-9d690180-22bd-11eb-8da4-167c7a6a2b2b.png)

### Actual behavior

For my sample of stars, the **mean distance** between the astropy code and any other of the 3 methods is **0.043 mas** whereas the mean distance between any of the 3 other methods is less than 5e-8mas

### Steps to Reproduce

Here is an example for a high proper-motion star ( Gaia DR2 source_id= 1639128849240549504)
1. Compute position at epoch 2000 in astropy
```python

from astropy.coordinates import SkyCoord

import astropy.units as u

from astropy.time import Time

from astropy.coordinates import Distance

ra = 264.3643538254156
dec = 71.06411252570835
parallax = 27.814599373682075
pmra = 245.46120536905048
pmdec = -407.9968194904302
radial_velocity = -46.641101097645986

c = SkyCoord(ra=ra*u.degree, dec=dec*u.degree, frame='icrs', pm_ra_cosdec= pmra*u.mas/u.yr, pm_dec = pmdec*u.mas/u.yr, obstime = Time(2015.5, format='decimalyear'), distance=Distance(parallax=parallax*u.mas), radial_velocity=radial_velocity*u.km/u.s)
coo_2000 = c.apply_space_motion(Time(2000.0, format='decimalyear'))
```
2. Compute position at epoch 2000 from ESA TAP service ( https://gea.esac.esa.int/archive/ ) using the following query :

```
SELECT EPOCH_PROP(ra, dec, parallax, pmra, pmdec, radial_velocity, 2015.5, 2000) 
FROM gaiadr2.gaia_source
WHERE source_id = 1639128849240549504
```
The result is :
`264.36109685407564,71.06586911414367,27.814027363024607,245.4730458523917,-407.96684044182194,-273.6769766141817`

3. Compute distance between position computed by astropy and position computed by ESA TAP:

```
c_esa = SkyCoord(ra=264.36109685407564*u.degree, dec=71.06586911414367*u.degree, frame='icrs')

coo_2000.separation(c_esa).mas
```

This outputs a distance of **0.16mas**

### System Details

Darwin-17.7.0-x86_64-i386-64bit
Python 3.6.7 |Anaconda, Inc.| (default, Oct 23 2018, 14:01:38) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
Numpy 1.19.0
astropy 4.0.1.post1
Scipy 1.3.1
Matplotlib 3.3.2

**Comment by eerovaher** [comment]
It might be worth pointing out that if `format='decimalyear'` is replaced with `format='jyear'` (as the ESA epoch propagation service documentation suggests should be done) the disagreement of the given example drops from 0.16 mas to 1.2e-5 mas.

**Comment by tboch** [comment]
Thanks @eerovaher ! I blindly copied the example on the page https://docs.astropy.org/en/stable/coordinates/apply_space_motion.html for TGAS epoch propagation, and applied to Gaia DR2.

**Comment by adrn** [comment]
This seems like a documentation bug! Thanks for catching this @tboch.

**Comment by mhvk** [comment]
Nice catch @eerovaher! `decimalyear` is a strange format in that the years do not have equal length...

**Comment by adrn** [comment]
Note: this will be closed by #11023 (if that gets merged), so do please have a look @tboch and @eerovaher if you have any comments!

**Comment by eerovaher** [comment]
The reference epochs of all Gaia data releases are in Barycentric Coordinate Time (TCB), but `Time` uses Coordinated Universal Time (UTC) by default. The difference between the two time scales is quite small, at least as far as epoch propagation is concerned, but it's still better to use the right time scale in the examples in the documentation, so I've opened #11722 to address this.
